### ISSUE TITLE
Fix support for 7.1 or improve detection

### ISSUE THREAD

**Issue by bwaidelich** [neos/neos-development-collection#1301]
### Description

When using Neos 2.3 with PHP 7.1 I got some weird (and very hard to debug) behavior:

The front page shows a 500 error, the following exception is logged
```
Exception #1180547966 in line 280 of Cache/Code/Flow_Object_Classes/TYPO3_TypoScript_Core_Parser.php: Syntax error in line 546. (2)
```
In the exception log, the backtrace shows

```
23 TYPO3\TypoScript\Core\Parser_Original::parseTypoScriptLine("2")
22 TYPO3\TypoScript\Core\Parser_Original::parse("prototype(TYPO3.Neos:Shortcut) < prototype(TYPO3.Neos:Document) {
    templatePath = 'resource://TYPO3.Neos/Private/Templates/NodeTypes/Shortcut.html'
    title = ${q(node).property("title")}
    uriPathSegment = ${q(node).property("uriPathSegment")}
    targetMode = ${q(node).property("targetMode")}

[...]

include: resource://TYPO3.Neos/Private/TypoScript/Root.ts2
include: resource://TYPO3.Neos.Seo/Private/TypoScript/Root.ts2

2

", "resource://My.Site/Private/TypoScript/Root.ts2")
21 TYPO3\Neos\Domain\Service\TypoScriptService_Original::getMergedTypoScriptObjectTree(TYPO3\TYPO3CR\Domain\Model\Node)

[...]

```

Browsing to the login screen just renders a blank page without any PHP or Flow error being logged.

### Affected Versions

Neos: 
2.3.6 (the only version I checked)

Flow: 
3.3.4 (the only version I checked)

Platform:
Heroku, nginx

**Comment by kdambekalns** [comment]
I guess this will be fixed with https://github.com/neos/flow-development-collection/pull/821 backported to Flow 3.3…

**Comment by bwaidelich** [comment]
I can confirm that neos/flow-development-collection#821 solves the issue for me (on Flow 3.3). But it's not yet backported, right?

**Comment by kdambekalns** [comment]
Great! No, not backported yet, AFAIK

**Comment by bwaidelich** [comment]
Fixed with  neos/flow-development-collection#936 (presumably)
